src/Components/CloudAccounts/CloudAccountsTable.tsx

```diff
@@ -0,0 +1,122 @@
+import React, { useEffect } from 'react';
+import {
+  SortByDirection,
+  Table,
+  Tbody,
+  Td,
+  Th,
+  Thead,
+  Tr,
+} from '@patternfly/react-table';
+import { Button, Content } from '@patternfly/react-core';
+import { useTableSort } from '../../hooks/util/tables/useTableSort';
+import { CloudAccountRow } from './types';
+import { CloudAccount } from '../../hooks/api/useCloudAccounts';
+import { CloudAccountStatus, getStatusIcon } from './GetStatusIcon';
+import { formatDate } from '../../hooks/util/dates';
+import { CloudAccountsPaginationData } from '../../state/cloudAccounts';
+import {
+  generateQueryParamsForData,
+  useQueryParamInformedAtom,
+} from '../../hooks/util/useQueryParam';
+import { Link } from 'react-router-dom';
+import {
+  shortNameToDisplay,
+  shortToFriendly,
+} from '../../hooks/util/cloudProviderMaps';
+
+type CloudAccountProps = {
+  cloudAccounts: CloudAccount[];
+};
+
+export const CloudAccountsTable = ({ cloudAccounts }: CloudAccountProps) => {
+  const [pagination, setPagination] = useQueryParamInformedAtom(
+    CloudAccountsPaginationData,
+    'pagination'
+  );
+
+  // const { page, perPage } = pagination;
+
+  useEffect(() => {
+    setPagination({
+      ...pagination,
+      itemCount: cloudAccounts.length,
+    });
+  }, [cloudAccounts.length]);
+
+  const rows: CloudAccountRow[] = cloudAccounts.map((acct) => ({
+    id: acct.providerAccountID,
+    provider: shortToFriendly[acct.shortName],
+    providerLabel: shortNameToDisplay[acct.shortName],
+    goldImage: acct.goldImageAccess as CloudAccountStatus,
+    date: acct.dateAdded,
+  }));
+
+  const { sorted, getSortParams } = useTableSort(rows, 'cloudAccounts', {
+    rowTranslator: (row) => [row.id, row.provider, row.goldImage, row.date],
+    initialSort: {
+      index: 0,
+      dir: SortByDirection.asc,
+    },
+  });
```

Comment:
Issue: This only sorts the displayed items, not all of the items. The API allows you to request sorted data from it so that the items can be sorted across pages.